ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -227,13 +225,37 @@ private[server] object ServerHelpers {
                 requestVault)
             }
 
+            // TODO: fix the order of error handling?
             result.attempt.flatMap {
               case Right((req, resp, drain)) =>
-                send(socket)(Some(req), resp, idleTimeout, onWriteFailure) >>
-                  drain.map {
-                    case Some(nextBuffer) => Some(((req, resp), (nextBuffer, true)))
-                    case None => None
-                  }
+                // TODO: Should we pay this cost for every HTTP request?
```

Comment:
This is why blaze has an enableWebSockets flag.  But how expensive _are_ vault lookups?